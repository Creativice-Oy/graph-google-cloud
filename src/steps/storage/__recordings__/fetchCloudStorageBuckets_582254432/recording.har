{
  "log": {
    "_recordingName": "fetchCloudStorageBuckets",
    "creator": {
      "comment": "persister:JupiterOneIntegationFSPersister",
      "name": "Polly.JS",
      "version": "4.3.0"
    },
    "entries": [
      {
        "_id": "acea721c8193b51ced888cae721cc423",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 739,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "content-type",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "content-length",
              "value": "739"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip,deflate"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "www.googleapis.com"
            }
          ],
          "headersSize": 283,
          "httpVersion": "HTTP/1.1",
          "method": "POST",
          "postData": {
            "mimeType": "application/json",
            "params": [],
            "text": "[REDACTED]"
          },
          "queryString": [],
          "url": "https://www.googleapis.com/oauth2/v4/token"
        },
        "response": {
          "bodySize": 597,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 597,
            "text": "{\"access_token\":\"[REDACTED]\",\"expires_in\":9999,\"token_type\":\"Bearer\"}"
          },
          "cookies": [],
          "headers": [
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin, Referer"
            },
            {
              "name": "date",
              "value": "Fri, 19 Feb 2021 14:40:50 GMT"
            },
            {
              "name": "server",
              "value": "scaffolding on HTTPServer2"
            },
            {
              "name": "cache-control",
              "value": "private"
            },
            {
              "name": "x-xss-protection",
              "value": "0"
            },
            {
              "name": "x-frame-options",
              "value": "SAMEORIGIN"
            },
            {
              "name": "x-content-type-options",
              "value": "nosniff"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 511,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-02-19T14:40:50.140Z",
        "time": 216,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 216
        }
      },
      {
        "_id": "d363efbccca866077514324c8f2c2baa",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/4.4.3 gl-node/14.15.4 auth/6.0.5"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/4.4.3 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "storage.googleapis.com"
            }
          ],
          "headersSize": 545,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [
            {
              "name": "project",
              "value": "j1-gc-integration-dev-v2"
            }
          ],
          "url": "https://storage.googleapis.com/storage/v1/b?project=j1-gc-integration-dev-v2"
        },
        "response": {
          "bodySize": 3312,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 3312,
            "text": "{\n  \"kind\": \"storage#buckets\",\n  \"items\": [\n    {\n      \"kind\": \"storage#bucket\",\n      \"selfLink\": \"https://www.googleapis.com/storage/v1/b/gcf-sources-711888229551-us-central1\",\n      \"id\": \"gcf-sources-711888229551-us-central1\",\n      \"name\": \"gcf-sources-711888229551-us-central1\",\n      \"projectNumber\": \"711888229551\",\n      \"metageneration\": \"1\",\n      \"location\": \"US-CENTRAL1\",\n      \"storageClass\": \"STANDARD\",\n      \"etag\": \"CAE=\",\n      \"timeCreated\": \"2021-02-18T00:04:30.131Z\",\n      \"updated\": \"2021-02-18T00:04:30.131Z\",\n      \"cors\": [\n        {\n          \"origin\": [\n            \"https://*.cloud.google.com\",\n            \"https://*.corp.google.com\",\n            \"https://*.corp.google.com:*\"\n          ],\n          \"method\": [\n            \"GET\"\n          ]\n        }\n      ],\n      \"iamConfiguration\": {\n        \"bucketPolicyOnly\": {\n          \"enabled\": true,\n          \"lockedTime\": \"2021-05-19T00:04:30.131Z\"\n        },\n        \"uniformBucketLevelAccess\": {\n          \"enabled\": true,\n          \"lockedTime\": \"2021-05-19T00:04:30.131Z\"\n        }\n      },\n      \"locationType\": \"region\"\n    },\n    {\n      \"kind\": \"storage#bucket\",\n      \"selfLink\": \"https://www.googleapis.com/storage/v1/b/j1-gc-integration-dev-v2-sink-logging-bucket\",\n      \"id\": \"j1-gc-integration-dev-v2-sink-logging-bucket\",\n      \"name\": \"j1-gc-integration-dev-v2-sink-logging-bucket\",\n      \"projectNumber\": \"711888229551\",\n      \"metageneration\": \"1\",\n      \"location\": \"US\",\n      \"storageClass\": \"STANDARD\",\n      \"etag\": \"CAE=\",\n      \"timeCreated\": \"2021-02-18T00:31:30.869Z\",\n      \"updated\": \"2021-02-18T00:31:30.869Z\",\n      \"iamConfiguration\": {\n        \"bucketPolicyOnly\": {\n          \"enabled\": false\n        },\n        \"uniformBucketLevelAccess\": {\n          \"enabled\": false\n        }\n      },\n      \"locationType\": \"multi-region\"\n    },\n    {\n      \"kind\": \"storage#bucket\",\n      \"selfLink\": \"https://www.googleapis.com/storage/v1/b/j1-gc-integration-dev-v2cloudfunctions\",\n      \"id\": \"j1-gc-integration-dev-v2cloudfunctions\",\n      \"name\": \"j1-gc-integration-dev-v2cloudfunctions\",\n      \"projectNumber\": \"711888229551\",\n      \"metageneration\": \"1\",\n      \"location\": \"US\",\n      \"storageClass\": \"STANDARD\",\n      \"etag\": \"CAE=\",\n      \"timeCreated\": \"2021-02-17T23:05:05.322Z\",\n      \"updated\": \"2021-02-17T23:05:05.322Z\",\n      \"iamConfiguration\": {\n        \"bucketPolicyOnly\": {\n          \"enabled\": false\n        },\n        \"uniformBucketLevelAccess\": {\n          \"enabled\": false\n        }\n      },\n      \"locationType\": \"multi-region\"\n    },\n    {\n      \"kind\": \"storage#bucket\",\n      \"selfLink\": \"https://www.googleapis.com/storage/v1/b/us.artifacts.j1-gc-integration-dev-v2.appspot.com\",\n      \"id\": \"us.artifacts.j1-gc-integration-dev-v2.appspot.com\",\n      \"name\": \"us.artifacts.j1-gc-integration-dev-v2.appspot.com\",\n      \"projectNumber\": \"711888229551\",\n      \"metageneration\": \"1\",\n      \"location\": \"US\",\n      \"storageClass\": \"STANDARD\",\n      \"etag\": \"CAE=\",\n      \"timeCreated\": \"2021-02-18T00:25:22.127Z\",\n      \"updated\": \"2021-02-18T00:25:22.127Z\",\n      \"iamConfiguration\": {\n        \"bucketPolicyOnly\": {\n          \"enabled\": false\n        },\n        \"uniformBucketLevelAccess\": {\n          \"enabled\": false\n        }\n      },\n      \"locationType\": \"multi-region\"\n    }\n  ]\n}\n"
          },
          "cookies": [],
          "headers": [
            {
              "name": "x-guploader-uploadid",
              "value": "[REDACTED]"
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "date",
              "value": "Fri, 19 Feb 2021 14:40:50 GMT"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin"
            },
            {
              "name": "cache-control",
              "value": "private, max-age=0, must-revalidate, no-transform"
            },
            {
              "name": "expires",
              "value": "Fri, 19 Feb 2021 14:40:50 GMT"
            },
            {
              "name": "content-length",
              "value": "3312"
            },
            {
              "name": "server",
              "value": "UploadServer"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 571,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-02-19T14:40:50.394Z",
        "time": 453,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 453
        }
      },
      {
        "_id": "309d186b6bd65e080120054707918537",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/4.4.3 gl-node/14.15.4 auth/6.0.5"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/4.4.3 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "storage.googleapis.com"
            }
          ],
          "headersSize": 553,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://storage.googleapis.com/storage/v1/b/gcf-sources-711888229551-us-central1/iam"
        },
        "response": {
          "bodySize": 812,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 812,
            "text": "{\n  \"kind\": \"storage#policy\",\n  \"resourceId\": \"projects/_/buckets/gcf-sources-711888229551-us-central1\",\n  \"version\": 1,\n  \"etag\": \"CAE=\",\n  \"bindings\": [\n    {\n      \"role\": \"roles/storage.legacyBucketOwner\",\n      \"members\": [\n        \"projectEditor:j1-gc-integration-dev-v2\",\n        \"projectOwner:j1-gc-integration-dev-v2\"\n      ]\n    },\n    {\n      \"role\": \"roles/storage.legacyBucketReader\",\n      \"members\": [\n        \"projectViewer:j1-gc-integration-dev-v2\"\n      ]\n    },\n    {\n      \"role\": \"roles/storage.legacyObjectOwner\",\n      \"members\": [\n        \"projectEditor:j1-gc-integration-dev-v2\",\n        \"projectOwner:j1-gc-integration-dev-v2\"\n      ]\n    },\n    {\n      \"role\": \"roles/storage.legacyObjectReader\",\n      \"members\": [\n        \"projectViewer:j1-gc-integration-dev-v2\"\n      ]\n    }\n  ]\n}\n"
          },
          "cookies": [],
          "headers": [
            {
              "name": "x-guploader-uploadid",
              "value": "[REDACTED]"
            },
            {
              "name": "etag",
              "value": "CAE="
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "date",
              "value": "Fri, 19 Feb 2021 14:40:51 GMT"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin"
            },
            {
              "name": "cache-control",
              "value": "private, max-age=0, must-revalidate, no-transform"
            },
            {
              "name": "expires",
              "value": "Fri, 19 Feb 2021 14:40:51 GMT"
            },
            {
              "name": "content-length",
              "value": "812"
            },
            {
              "name": "server",
              "value": "UploadServer"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 582,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-02-19T14:40:50.853Z",
        "time": 326,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 326
        }
      },
      {
        "_id": "91b316d87f00710bf4428623136df86a",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/4.4.3 gl-node/14.15.4 auth/6.0.5"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/4.4.3 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "storage.googleapis.com"
            }
          ],
          "headersSize": 561,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://storage.googleapis.com/storage/v1/b/j1-gc-integration-dev-v2-sink-logging-bucket/iam"
        },
        "response": {
          "bodySize": 494,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 494,
            "text": "{\n  \"kind\": \"storage#policy\",\n  \"resourceId\": \"projects/_/buckets/j1-gc-integration-dev-v2-sink-logging-bucket\",\n  \"version\": 1,\n  \"etag\": \"CAE=\",\n  \"bindings\": [\n    {\n      \"role\": \"roles/storage.legacyBucketOwner\",\n      \"members\": [\n        \"projectEditor:j1-gc-integration-dev-v2\",\n        \"projectOwner:j1-gc-integration-dev-v2\"\n      ]\n    },\n    {\n      \"role\": \"roles/storage.legacyBucketReader\",\n      \"members\": [\n        \"projectViewer:j1-gc-integration-dev-v2\"\n      ]\n    }\n  ]\n}\n"
          },
          "cookies": [],
          "headers": [
            {
              "name": "x-guploader-uploadid",
              "value": "[REDACTED]"
            },
            {
              "name": "etag",
              "value": "CAE="
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "date",
              "value": "Fri, 19 Feb 2021 14:40:51 GMT"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin"
            },
            {
              "name": "cache-control",
              "value": "private, max-age=0, must-revalidate, no-transform"
            },
            {
              "name": "expires",
              "value": "Fri, 19 Feb 2021 14:40:51 GMT"
            },
            {
              "name": "content-length",
              "value": "494"
            },
            {
              "name": "server",
              "value": "UploadServer"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 582,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-02-19T14:40:51.194Z",
        "time": 342,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 342
        }
      },
      {
        "_id": "caba15abb65e4142694b8a1e0609103a",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/4.4.3 gl-node/14.15.4 auth/6.0.5"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/4.4.3 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "storage.googleapis.com"
            }
          ],
          "headersSize": 555,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://storage.googleapis.com/storage/v1/b/j1-gc-integration-dev-v2cloudfunctions/iam"
        },
        "response": {
          "bodySize": 488,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 488,
            "text": "{\n  \"kind\": \"storage#policy\",\n  \"resourceId\": \"projects/_/buckets/j1-gc-integration-dev-v2cloudfunctions\",\n  \"version\": 1,\n  \"etag\": \"CAE=\",\n  \"bindings\": [\n    {\n      \"role\": \"roles/storage.legacyBucketOwner\",\n      \"members\": [\n        \"projectEditor:j1-gc-integration-dev-v2\",\n        \"projectOwner:j1-gc-integration-dev-v2\"\n      ]\n    },\n    {\n      \"role\": \"roles/storage.legacyBucketReader\",\n      \"members\": [\n        \"projectViewer:j1-gc-integration-dev-v2\"\n      ]\n    }\n  ]\n}\n"
          },
          "cookies": [],
          "headers": [
            {
              "name": "x-guploader-uploadid",
              "value": "[REDACTED]"
            },
            {
              "name": "etag",
              "value": "CAE="
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "date",
              "value": "Fri, 19 Feb 2021 14:40:51 GMT"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin"
            },
            {
              "name": "cache-control",
              "value": "private, max-age=0, must-revalidate, no-transform"
            },
            {
              "name": "expires",
              "value": "Fri, 19 Feb 2021 14:40:51 GMT"
            },
            {
              "name": "content-length",
              "value": "488"
            },
            {
              "name": "server",
              "value": "UploadServer"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 582,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-02-19T14:40:51.540Z",
        "time": 397,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 397
        }
      },
      {
        "_id": "2fb3a4ce13725306b730ba1780a214f1",
        "_order": 0,
        "cache": {},
        "request": {
          "bodySize": 0,
          "cookies": [],
          "headers": [
            {
              "_fromType": "array",
              "name": "x-goog-api-client",
              "value": "gdcl/4.4.3 gl-node/14.15.4 auth/6.0.5"
            },
            {
              "_fromType": "array",
              "name": "accept-encoding",
              "value": "gzip"
            },
            {
              "_fromType": "array",
              "name": "user-agent",
              "value": "google-api-nodejs-client/4.4.3 (gzip)"
            },
            {
              "_fromType": "array",
              "name": "authorization",
              "value": "[REDACTED]"
            },
            {
              "_fromType": "array",
              "name": "accept",
              "value": "application/json"
            },
            {
              "_fromType": "array",
              "name": "connection",
              "value": "close"
            },
            {
              "name": "host",
              "value": "storage.googleapis.com"
            }
          ],
          "headersSize": 566,
          "httpVersion": "HTTP/1.1",
          "method": "GET",
          "queryString": [],
          "url": "https://storage.googleapis.com/storage/v1/b/us.artifacts.j1-gc-integration-dev-v2.appspot.com/iam"
        },
        "response": {
          "bodySize": 499,
          "content": {
            "mimeType": "application/json; charset=UTF-8",
            "size": 499,
            "text": "{\n  \"kind\": \"storage#policy\",\n  \"resourceId\": \"projects/_/buckets/us.artifacts.j1-gc-integration-dev-v2.appspot.com\",\n  \"version\": 1,\n  \"etag\": \"CAE=\",\n  \"bindings\": [\n    {\n      \"role\": \"roles/storage.legacyBucketOwner\",\n      \"members\": [\n        \"projectEditor:j1-gc-integration-dev-v2\",\n        \"projectOwner:j1-gc-integration-dev-v2\"\n      ]\n    },\n    {\n      \"role\": \"roles/storage.legacyBucketReader\",\n      \"members\": [\n        \"projectViewer:j1-gc-integration-dev-v2\"\n      ]\n    }\n  ]\n}\n"
          },
          "cookies": [],
          "headers": [
            {
              "name": "x-guploader-uploadid",
              "value": "[REDACTED]"
            },
            {
              "name": "etag",
              "value": "CAE="
            },
            {
              "name": "content-type",
              "value": "application/json; charset=UTF-8"
            },
            {
              "name": "date",
              "value": "Fri, 19 Feb 2021 14:40:52 GMT"
            },
            {
              "name": "vary",
              "value": "Origin, X-Origin"
            },
            {
              "name": "cache-control",
              "value": "private, max-age=0, must-revalidate, no-transform"
            },
            {
              "name": "expires",
              "value": "Fri, 19 Feb 2021 14:40:52 GMT"
            },
            {
              "name": "content-length",
              "value": "499"
            },
            {
              "name": "server",
              "value": "UploadServer"
            },
            {
              "name": "alt-svc",
              "value": "h3-29=\":443\"; ma=2592000,h3-T051=\":443\"; ma=2592000,h3-Q050=\":443\"; ma=2592000,h3-Q046=\":443\"; ma=2592000,h3-Q043=\":443\"; ma=2592000,quic=\":443\"; ma=2592000; v=\"46,43\""
            },
            {
              "name": "connection",
              "value": "close"
            }
          ],
          "headersSize": 582,
          "httpVersion": "HTTP/1.1",
          "redirectURL": "",
          "status": 200,
          "statusText": "OK"
        },
        "startedDateTime": "2021-02-19T14:40:51.940Z",
        "time": 297,
        "timings": {
          "blocked": -1,
          "connect": -1,
          "dns": -1,
          "receive": 0,
          "send": 0,
          "ssl": -1,
          "wait": 297
        }
      }
    ],
    "pages": [],
    "version": "1.2"
  }
}
