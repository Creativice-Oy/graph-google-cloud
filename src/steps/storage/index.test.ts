import {
  Recording,
  createMockStepExecutionContext,
} from '@jupiterone/integration-sdk-testing';
import { setupGoogleCloudRecording } from '../../../test/recording';
import { IntegrationConfig } from '../../types';
import { fetchStorageBuckets } from '.';
import { integrationConfig } from '../../../test/config';
import {
  CLOUD_STORAGE_BUCKET_ENTITY_CLASS,
  CLOUD_STORAGE_BUCKET_ENTITY_TYPE,
} from './constants';

describe('#fetchCloudStorageBuckets', () => {
  let recording: Recording;

  beforeEach(() => {
    recording = setupGoogleCloudRecording({
      directory: __dirname,
      name: 'fetchCloudStorageBuckets',
    });
  });

  afterEach(async () => {
    await recording.stop();
  });

  test('should collect data', async () => {
    const context = createMockStepExecutionContext<IntegrationConfig>({
      instanceConfig: integrationConfig,
    });

    await fetchStorageBuckets(context);

    expect(context.jobState.collectedRelationships.length).toEqual(0);
    expect(context.jobState.collectedEntities.length).toBeGreaterThanOrEqual(1);

    expect(context.jobState.collectedEntities).toEqual(
      context.jobState.collectedEntities.map((e) =>
        expect.objectContaining({
          ...e,
          _rawData: expect.any(Array),
          _class: [CLOUD_STORAGE_BUCKET_ENTITY_CLASS],
          _type: CLOUD_STORAGE_BUCKET_ENTITY_TYPE,
          _key: `bucket:${e.id}`,
          name: expect.any(String),
          displayName: e.name,
        }),
      ),
    );
  });
});
