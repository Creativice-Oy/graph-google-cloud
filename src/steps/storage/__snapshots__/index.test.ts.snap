// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#fetchCloudStorageBuckets should collect data 1`] = `
Object {
  "collectedEntities": Array [
    Object {
      "_class": Array [
        "DataStore",
      ],
      "_key": "bucket:customer-managed-encryption-key-bucket-545240943112",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "encryption": Object {
              "defaultKmsKeyName": "projects/j1-gc-integration-dev/locations/us/keyRings/j1-gc-integration-dev-bucket-ring/cryptoKeys/j1-gc-integration-dev-bucket-key",
            },
            "etag": "CAE=",
            "iamConfiguration": Object {
              "bucketPolicyOnly": Object {
                "enabled": false,
              },
              "uniformBucketLevelAccess": Object {
                "enabled": false,
              },
            },
            "id": "customer-managed-encryption-key-bucket-545240943112",
            "kind": "storage#bucket",
            "location": "US",
            "locationType": "multi-region",
            "metageneration": "1",
            "name": "customer-managed-encryption-key-bucket-545240943112",
            "projectNumber": "545240943112",
            "selfLink": "https://www.googleapis.com/storage/v1/b/customer-managed-encryption-key-bucket-545240943112",
            "storageClass": "STANDARD",
            "timeCreated": "2020-07-28T19:06:14.033Z",
            "updated": "2020-07-28T19:06:14.033Z",
          },
        },
      ],
      "_type": "google_storage_bucket",
      "classification": null,
      "createdOn": 1595963174033,
      "displayName": "customer-managed-encryption-key-bucket-545240943112",
      "encrypted": true,
      "encryptionKeyRef": "projects/j1-gc-integration-dev/locations/us/keyRings/j1-gc-integration-dev-bucket-ring/cryptoKeys/j1-gc-integration-dev-bucket-key",
      "etag": "CAE=",
      "id": "customer-managed-encryption-key-bucket-545240943112",
      "location": "US",
      "name": "customer-managed-encryption-key-bucket-545240943112",
      "public": false,
      "retentionDate": undefined,
      "retentionPeriod": undefined,
      "retentionPolicyEnabled": undefined,
      "storageClass": "STANDARD",
      "uniformBucketLevelAccess": false,
      "updatedOn": 1595963174033,
      "webLink": "https://console.cloud.google.com/storage/browser/customer-managed-encryption-key-bucket-545240943112;tab=objects?forceOnBucketsSortingFiltering=false&project=j1-gc-integration-dev",
    },
    Object {
      "_class": Array [
        "DataStore",
      ],
      "_key": "bucket:j1-gc-integration-devcloudfunctions",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "etag": "CAE=",
            "iamConfiguration": Object {
              "bucketPolicyOnly": Object {
                "enabled": false,
              },
              "uniformBucketLevelAccess": Object {
                "enabled": false,
              },
            },
            "id": "j1-gc-integration-devcloudfunctions",
            "kind": "storage#bucket",
            "location": "US",
            "locationType": "multi-region",
            "metageneration": "1",
            "name": "j1-gc-integration-devcloudfunctions",
            "projectNumber": "545240943112",
            "selfLink": "https://www.googleapis.com/storage/v1/b/j1-gc-integration-devcloudfunctions",
            "storageClass": "STANDARD",
            "timeCreated": "2020-07-28T18:34:48.723Z",
            "updated": "2020-07-28T18:34:48.723Z",
          },
        },
      ],
      "_type": "google_storage_bucket",
      "classification": null,
      "createdOn": 1595961288723,
      "displayName": "j1-gc-integration-devcloudfunctions",
      "encrypted": true,
      "encryptionKeyRef": undefined,
      "etag": "CAE=",
      "id": "j1-gc-integration-devcloudfunctions",
      "location": "US",
      "name": "j1-gc-integration-devcloudfunctions",
      "public": false,
      "retentionDate": undefined,
      "retentionPeriod": undefined,
      "retentionPolicyEnabled": undefined,
      "storageClass": "STANDARD",
      "uniformBucketLevelAccess": false,
      "updatedOn": 1595961288723,
      "webLink": "https://console.cloud.google.com/storage/browser/j1-gc-integration-devcloudfunctions;tab=objects?forceOnBucketsSortingFiltering=false&project=j1-gc-integration-dev",
    },
    Object {
      "_class": Array [
        "DataStore",
      ],
      "_key": "bucket:test-public-storage-bucket-545240943112",
      "_rawData": Array [
        Object {
          "name": "default",
          "rawData": Object {
            "etag": "CAI=",
            "iamConfiguration": Object {
              "bucketPolicyOnly": Object {
                "enabled": false,
              },
              "uniformBucketLevelAccess": Object {
                "enabled": false,
              },
            },
            "id": "test-public-storage-bucket-545240943112",
            "kind": "storage#bucket",
            "location": "US",
            "locationType": "multi-region",
            "metageneration": "2",
            "name": "test-public-storage-bucket-545240943112",
            "projectNumber": "545240943112",
            "selfLink": "https://www.googleapis.com/storage/v1/b/test-public-storage-bucket-545240943112",
            "storageClass": "STANDARD",
            "timeCreated": "2020-10-08T18:54:42.000Z",
            "updated": "2020-10-08T18:54:43.402Z",
          },
        },
      ],
      "_type": "google_storage_bucket",
      "classification": null,
      "createdOn": 1602183282000,
      "displayName": "test-public-storage-bucket-545240943112",
      "encrypted": true,
      "encryptionKeyRef": undefined,
      "etag": "CAI=",
      "id": "test-public-storage-bucket-545240943112",
      "location": "US",
      "name": "test-public-storage-bucket-545240943112",
      "public": true,
      "retentionDate": undefined,
      "retentionPeriod": undefined,
      "retentionPolicyEnabled": undefined,
      "storageClass": "STANDARD",
      "uniformBucketLevelAccess": false,
      "updatedOn": 1602183283402,
      "webLink": "https://console.cloud.google.com/storage/browser/test-public-storage-bucket-545240943112;tab=objects?forceOnBucketsSortingFiltering=false&project=j1-gc-integration-dev",
    },
  ],
  "collectedRelationships": Array [],
  "encounteredTypes": Array [
    "google_storage_bucket",
  ],
  "numCollectedEntities": 3,
  "numCollectedRelationships": 0,
}
`;
